streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -271,28 +271,30 @@ private long processNext(final List<ConsumerRecord<byte[], byte[]>> records,
                              final StateRestorer restorer,
                              final Long endOffset) {
         final List<KeyValue<byte[], byte[]>> restoreRecords = new ArrayList<>();
-        long offset = -1;
+        long nextPosition = -1;
 
         for (final ConsumerRecord<byte[], byte[]> record : records) {
-            offset = record.offset();
+            final long offset = record.offset();
             if (restorer.hasCompleted(offset, endOffset)) {
+                nextPosition = record.offset();
                 break;
             }
             if (record.key() != null) {
                 restoreRecords.add(KeyValue.pair(record.key(), record.value()));
             }
         }
 
-        if (offset == -1) {
-            offset = restoreConsumer.position(restorer.partition());
+        if (nextPosition == -1) {
+            nextPosition = restoreConsumer.position(restorer.partition());
         }
 
         if (!restoreRecords.isEmpty()) {
             restorer.restore(restoreRecords);
-            restorer.restoreBatchCompleted(offset + 1, records.size());
+            restorer.restoreBatchCompleted(nextPosition, records.size());
+
```

Comment:
nit: remove empty line

As discussed with @guozhangwang offline, I think passing in `nextPosition` is not correct. We should pass in the offset of the latest restored offset, but `nextPosition` is the offset of the first not restored offset.

Question is, if a `nextPosition - 1` would be desirable or not because this might be a commit marker position and not an actual record position.

Reply:
Correct, this will be handled in a follow-up PR

Reply:
Ack. Can you create a JIRA for it?

Reply:
Done - https://issues.apache.org/jira/browse/KAFKA-6367